src/pool/srv_target.c

```diff
@@ -990,8 +992,33 @@ ds_pool_tgt_map_update(struct ds_pool *pool, struct pool_buf *buf,
 		rc = dss_task_collective(update_child_map, pool, 0,
 					 DSS_ULT_IO);
 		D_ASSERT(rc == 0);
+		update_map = true;
 	}
 
+	if (update_map) {
+		struct dtx_resync_arg *arg;
+		int ret;
+
+		/* Since the map has been updated successfully, so let's
+		 * ignore the dtx resync failure for now.
+		 */
+		D_ALLOC_PTR(arg);
+		if (arg == NULL)
+			D_GOTO(out, rc);
+
+		uuid_copy(arg->pool_uuid, pool->sp_uuid);
+		arg->version = pool->sp_map_version;
+		ret = dss_ult_create(dtx_resync_ult, arg, DSS_ULT_POOL_SRV,
+				    0, 0, NULL);
```

Comment:
The original logic is that the build process will trigger dtx_resync and wait until dtx_resync done, otherwise, the subsequent rebuild logic may be blocked or miss some entries. So here, with your new patch, handle the DTX resync via independent ULT, we need to guarantee that the subsequent rebuild logic will not be triggered until such ULT has done the dtx_resync.

Reply:
I thought about it, but really dislike the barrier here.  I think rebuild blocking should be ok (just retry) as long as it is not deadlock.  Could you please remind me the missing entries case here?  I wonder could we resolve that in different way. Thanks.

Reply:
In current implementation, the vos_dtx_check_availability() will return ALB_UNAVAILABLE if the DTX is 'PREPARED' (but the old leader has already marked it as committable or committed before corruption). Then during the scan for object rebuild, the object/shard with 'prepared' DTX  on the new leader will be skipped for rebuild.